orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/MonitoredDeployBaseTask.java

```diff
@@ -267,22 +263,21 @@ private MonitoredDeployStageData getStageContext(StageExecution stage) {
     return stage.mapTo(MonitoredDeployStageData.class);
   }
 
-  private String getRetrofitLogMessage(Response response) {
-    if (response == null) {
+  private String getRetrofitLogMessage(SpinnakerServerException spinnakerException) {
+    if (!(spinnakerException instanceof SpinnakerHttpException)) {
       return "<NO RESPONSE>";
     }
 
+    SpinnakerHttpException httpException = (SpinnakerHttpException) spinnakerException;
     String body = "";
     String status = "";
     String headers = "";
+    ObjectMapper objectMapper = new ObjectMapper();
 
     try {
-      status = String.format("%d (%s)", response.getStatus(), response.getReason());
-      body =
-          CharStreams.toString(
-              new InputStreamReader(response.getBody().in(), StandardCharsets.UTF_8));
-      headers =
-          response.getHeaders().stream().map(Header::toString).collect(Collectors.joining("\n"));
+      status = String.format("%d (%s)", httpException.getResponseCode(), httpException.getReason());
+      body = objectMapper.writeValueAsString(httpException.getResponseBody());
+      headers = httpException.getHeaders().toString();
     } catch (Exception e) {
       log.error(
           "Failed to fully parse retrofit error while reading response from deployment monitor", e);
```

Comment:
```suggestion
          "Failed to fully parse http response while reading response from deployment monitor", e);
```
and, is it correct to only wrap the objectMapper.writeValueAsString in the try/catch?

Reply:
Reading the status code from the exception instance is now removed and now we are leftover with 2 lines, reading the body and headers, So wrapping the line : httpException.getHeaders().toString(); in try/catch is anyway not gonna harm us and the statements outside the try/catch seems to less risky. 

hence keeping it the same way like it was before, and just going with the change in error log

Reply:
I'm trying to make sure the error message is correct.  If we get an exception dealing with the headers, then the message isn't correct.  It's not really about reading the response either at this point...it's about serializing the response body....or something...

Reply:
Please refer to this suggeston - https://github.com/spinnaker/orca/pull/4608/files/4c608efdc4f4783fd240be38142f4e1d3e374955#r1403955017